### PR TITLE
Consistency with format Part 2

### LINKED ISSUE #4870: Consistency with `format` in docstring
There are a number of format options available in various functions throughout the code. 
At times, it is not clear which options are available or if there is only currently one option. This can be confusing, especially for a new user. 

For example here:
https://github.com/gammapy/gammapy/blob/b34bbe35b0044ce4c68d2fce67801e600160515a/gammapy/maps/axes.py#L1092-L1120

Versus a well explained docstring:
https://github.com/gammapy/gammapy/blob/b34bbe35b0044ce4c68d2fce67801e600160515a/gammapy/maps/hpx/core.py#L104-L134

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This PR is part 2 to resolve #4870.

**Description**
It provides a unified format for various string options.

### CODE REVIEW

File: gammapy/maps/region/geom.py [hunk 1]
```diff
@@ -678,7 +678,7 @@ def to_hdulist(self, format="ogip", hdu_bands=None, hdu_region=None):
 
         Parameters
         ----------
-        format : {"gadf", "ogip", "ogip-sherpa"}, optional
+        format : {"gadf", "ogip", "ogip-sherpa"}
```
Comment (registerrier) [inline gammapy/maps/region/geom.py:681]:
```suggestion
        format : {"ogip", "gadf", "ogip-sherpa"}
```

File: gammapy/maps/hpx/core.py [hunk 2]
```diff
@@ -303,7 +303,7 @@ def to_hdu(self, hdu=None, hdu_bands=None, sparse=False, format=None):
             Set INDXSCHM to SPARSE and sparsify the map by only
             writing pixels with non-zero amplitude.
             Default is False.
-        format : {'fgst-ccube', 'fgst-template', 'gadf', None}, optional
+        format : {'fgst-ccube', 'fgst-template', 'gadf', None}
```
Comment (registerrier) [inline gammapy/maps/hpx/core.py:306]:
```suggestion
        format : {None, 'fgst-ccube', 'fgst-template', 'gadf'}
```

**Review by registerrier — COMMENTED** [review]
Thanks @Astro-Kirsty . This looks good.
My main concern regards the list of of possible choices, giving just the list of choices suggest s that the expected input is a single string among those provided.

**Comment by Astro-Kirsty** [comment]
> Thanks @Astro-Kirsty . This looks good. My main concern regards the list of of possible choices, giving just the list of choices suggest s that the expected input is a single string among those provided.

Yes, in general I agree too. The other option is to use the following, however, I feel sometimes the options get lost in the docstring that way:
image

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . Looks good!

**Pull request merged**